K8s-deployment/K8s-cluster/addons/mon-stack/loki/loki-values.yaml

```diff
@@ -1,282 +1,1109 @@
+          service: {{ .service }}
+    {{- end }}
+    {{- end }}
+
+    {{- if .Values.loki.schemaConfig}}
+    schema_config:
+    {{- toYaml .Values.loki.schemaConfig | nindent 2}}
+    {{- else }}
+    schema_config:
+      configs:
+        - from: 2022-01-11
+          store: boltdb-shipper
+          {{- if eq .Values.loki.storage.type "s3" }}
+          object_store: s3
+          {{- else if eq .Values.loki.storage.type "gcs" }}
+          object_store: gcs
+          {{- else }}
+          object_store: filesystem
+          {{- end }}
+          schema: v12
+          index:
+            prefix: loki_index_
+            period: 24h
+    {{- end }}
+
+    {{- if or .Values.minio.enabled (eq .Values.loki.storage.type "s3") (eq .Values.loki.storage.type "gcs") }}
+    ruler:
+      storage:
+      {{- include "loki.rulerStorageConfig" . | nindent 4}}
+    {{- end -}}
+
+    {{- with .Values.loki.memcached.results_cache }}
+    query_range:
+      align_queries_with_step: true
+      {{- if and .enabled .host }}
+      cache_results: {{ .enabled }}
+      results_cache:
+        cache:
+          default_validity: {{ .default_validity }}
+          memcached_client:
+            host: {{ .host }}
+            service: {{ .service }}
+            timeout: {{ .timeout }}
+      {{- end }}
+    {{- end }}
+
+    {{- with .Values.loki.storage_config }}
+    storage_config:
+      {{- tpl (. | toYaml) $ | nindent 4 }}
+    {{- end }}
+
+    {{- with .Values.loki.query_scheduler }}
+    query_scheduler:
+      {{- tpl (. | toYaml) $ | nindent 4 }}
+    {{- end }}
+
+    {{- with .Values.loki.compactor }}
+    compactor:
+      {{- tpl (. | toYaml) $ | nindent 4 }}
+    {{- end }}
+
+  # Should authentication be enabled
+  auth_enabled: true
+
+  # -- Check https://grafana.com/docs/loki/latest/configuration/#server for more info on the server configuration.
+  serverConfig:
+    http_listen_port: 3100
+    grpc_listen_port: 9095
+
+  # -- Limits config
   limits_config:
     enforce_metric_name: false
     reject_old_samples: true
     reject_old_samples_max_age: 168h
-    retention_period: 744h
-  schema_config:
-    configs:
-      - from: 2020-10-24
-        store: boltdb-shipper
-        object_store: filesystem
-        schema: v11
-        index:
-          prefix: index_
-          period: 24h
-  server:
-    http_listen_port: 3100
-  storage_config:
-    boltdb_shipper:
-      active_index_directory: /data/loki/boltdb-shipper-active
-      cache_location: /data/loki/boltdb-shipper-cache
-      cache_ttl: 24h         # Can be increased for faster performance over longer query periods, uses more disk space
-      shared_store: filesystem
+    max_cache_freshness_per_query: 10m
+    split_queries_by_interval: 15m
+
+  # -- Check https://grafana.com/docs/loki/latest/configuration/#common_config for more info on how to provide a common configuration
+  commonConfig:
+    path_prefix: /var/loki
+    replication_factor: 3
```

Comment:
need to use  ```replication_factor:1 ``` as we currently use file-system backed storage and not object storage like s3 .Ref: https://github.com/grafana/loki/tree/main/production/helm/loki#upgrading-from-grafanaloki